### PR TITLE
Add unit tests for converters/vmss

### LINKED ISSUE #2896: Add a unit test for SDKToVMSSVM function
The `SDKToVMSSVM` function is missing a unit test https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/converters/vmss.go#L65

Unit test should be added in https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/converters/vmss_test.go

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**:

/kind cleanup

**What this PR does / why we need it**:

A couple of functions in azure/converters/vmss.go lacked unit tests.

**Which issue(s) this PR fixes**:

Fixes #2896 

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
I believe this fixes #2896

**Review by CecileRobertMichon — COMMENTED** [review]
/assign @willie-yao @nawazkh

File: azure/converters/vmss_test.go [hunk 1]
```diff
@@ -122,3 +123,137 @@ func Test_SDKToVMSS(t *testing.T) {
 		})
 	}
 }
+
+func Test_SDKToVMSSVM(t *testing.T) {
```
Comment (willie-yao) [inline azure/converters/vmss_test.go:127]: Would it be helpful to add a test case for a nil `VirtualMachineScaleSetVMProperties`?

**Review by willie-yao — COMMENTED** [review]
Overall this looks good!! This may be outside the scope of this PR, but I think it'll be helpful to add a few more test cases to `Test_SDKToVMSS`, such as a minimal case.

**Comment by nawazkh** [comment]
looks good to me!

File: azure/converters/vmss_test.go (hunk 1 above)
Comment (mboersma) [reply azure/converters/vmss_test.go:127]: Good idea–we don't explicitly set it to `nil` in the tests, but that is a branch in the function being tested. Added.

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve
